ChatSecure/Classes/View Controllers/OTRMessagesViewController.m

```diff
@@ -358,6 +360,21 @@ - (nullable OTRAccount *)accountWithTransaction:(nonnull YapDatabaseReadTransact
     return account;
 }
 
+- (nullable OTRXMPPRoomOccupant *)occupantWithTransaction:(nonnull YapDatabaseReadTransaction *)transaction forFullJid:(nonnull NSString*)fullJid inRoom:(nonnull NSString *)roomUniqueId {
```

Comment:
This would be a lot faster with a secondary index. Right now it's basically O(n) for the number of occupants. Considering this method will be called repeatedly while the UI is scrolling around (and nothing is cached), I would not be surprised to see noticeable performance problems / stuttering for larger rooms. Also I think this helper function would be better moved to the `OTRXMPPRoomOccupant` or `OTRXMPPRoom` classes.

Admittedly it's a bit of a mess to set up the secondary indexes so I can handle this myself after the merge. Just something to note for future consideration, try to avoid blocking O(n) lookups in frequently called UI functions.